src/utils/ErrorHandler.ts

```diff
@@ -0,0 +1,32 @@
+interface IError {
+    statusCode: number;
+    error: string[];
+    data: any;
+    success: boolean;
+    provider: string;
+}
+
+class ErrorHandler extends Error implements IError {
+    public statusCode: number;
+    public error: string[];
+    public data: any;
+    public success: boolean;
+    public provider: string;
+
+    constructor( provider: string, message: string = "Something went wrong", statusCode: number = 500, error: string[] = [], stack?: string) {
```

Comment:
`provider` parameter should be of `Provider` type. You can import `Provider` type from src/interfaces/IProviderConfig.ts.
`constructor(provider: Provider, ...`